packages/core/src/utils/make-namespaced.ts

```diff
@@ -228,28 +228,39 @@ function toDummyEnumWithAstId(astId: number) {
 }
 
 function replaceFunction(node: FunctionDefinition, orig: Buffer, modifications: Modification[]) {
-  if (node.kind === 'constructor') {
-    modifications.push(makeDelete(node, orig));
-  } else {
-    if (node.modifiers.length > 0) {
-      for (const modifier of node.modifiers) {
-        modifications.push(makeDelete(modifier, orig));
+  switch (node.kind) {
+    case 'freeFunction':
+    case 'function': {
+      if (node.modifiers.length > 0) {
+        for (const modifier of node.modifiers) {
+          modifications.push(makeDelete(modifier, orig));
+        }
       }
-    }
 
-    if (node.returnParameters.parameters.length > 0) {
-      modifications.push(
-        makeReplace(
-          node.returnParameters,
-          orig,
-          `(${node.returnParameters.parameters.map(param => toReturnParameterReplacement(param)).join(', ')})`,
-        ),
-      );
-    }
+      if (node.returnParameters.parameters.length > 0) {
+        modifications.push(
+          makeReplace(
+            node.returnParameters,
+            orig,
+            `(${node.returnParameters.parameters.map(param => toReturnParameterReplacement(param)).join(', ')})`,
+          ),
+        );
+      }
+
+      if (node.body) {
+        modifications.push(makeReplace(node.body, orig, '{}'));
+      }
 
-    if (node.body) {
-      modifications.push(makeReplace(node.body, orig, '{}'));
+      break;
+    }
+    case 'constructor':
+    case 'fallback':
+    case 'receive': {
+      modifications.push(makeDelete(node, orig));
```

Comment:
Should we be concerned about orphaned NatSpec? Or do we consider that a solved issue now with #1051?

Reply:
Changed to replace with dummy enums.  Orphaned NatSpec is indeed possible on platforms that don't support Slang.